src/applications/personalization/profile360/vet360/actions/transactions.js

```diff
@@ -111,186 +97,41 @@ export function refreshTransaction(transaction, analyticsSectionName) {
+      recordEvent({
+        event: method === 'DELETE' ? 'profile-deleted' : 'profile-transaction',
+        'profile-section': analyticsSectionName
+      });
 
-        dispatch({
-          type: VET360_TRANSACTION_REQUEST_SUCCEEDED,
-          fieldName,
-          transaction
-        });
-      } catch (error) {
-        dispatch({
-          type: VET360_TRANSACTION_REQUEST_FAILED,
-          error,
-          fieldName
-        });
-      }
-    };
+      dispatch({
+        type: VET360_TRANSACTION_REQUEST_SUCCEEDED,
+        fieldName,
+        transaction
+      });
+    } catch (error) {
+      dispatch({
+        type: VET360_TRANSACTION_REQUEST_FAILED,
+        error,
+        fieldName
+      });
+    }
   };
 }
-
-export const fieldUpdaters = {
-  [VET360_CONSTANTS.FIELD_NAMES.HOME_PHONE]: {
-    update: updateVet360Field('/profile/telephones', VET360_CONSTANTS.FIELD_NAMES.HOME_PHONE, 'phone'),
-    destroy: deleteVet360Field('/profile/telephones', VET360_CONSTANTS.FIELD_NAMES.HOME_PHONE, 'phone')
-  },
-
-  [VET360_CONSTANTS.FIELD_NAMES.MOBILE_PHONE]: {
-    update: updateVet360Field('/profile/telephones', VET360_CONSTANTS.FIELD_NAMES.MOBILE_PHONE, 'phone'),
-    destroy: deleteVet360Field('/profile/telephones', VET360_CONSTANTS.FIELD_NAMES.MOBILE_PHONE, 'phone')
-  },
-
-  [VET360_CONSTANTS.FIELD_NAMES.WORK_PHONE]: {
-    update: updateVet360Field('/profile/telephones', VET360_CONSTANTS.FIELD_NAMES.WORK_PHONE, 'phone'),
-    destroy: deleteVet360Field('/profile/telephones', VET360_CONSTANTS.FIELD_NAMES.WORK_PHONE, 'phone')
-  },
-
-  [VET360_CONSTANTS.FIELD_NAMES.TEMP_PHONE]: {
-    update: updateVet360Field('/profile/telephones', VET360_CONSTANTS.FIELD_NAMES.TEMP_PHONE, 'phone'),
-    destroy: deleteVet360Field('/profile/telephones', VET360_CONSTANTS.FIELD_NAMES.TEMP_PHONE, 'phone')
-  },
-
-  [VET360_CONSTANTS.FIELD_NAMES.FAX_NUMBER]: {
-    update: updateVet360Field('/profile/telephones', VET360_CONSTANTS.FIELD_NAMES.FAX_NUMBER, 'phone'),
-    destroy: deleteVet360Field('/profile/telephones', VET360_CONSTANTS.FIELD_NAMES.FAX_NUMBER, 'phone')
-  },
-
-  [VET360_CONSTANTS.FIELD_NAMES.EMAIL]: {
-    update: updateVet360Field('/profile/email_addresses', VET360_CONSTANTS.FIELD_NAMES.EMAIL, 'email'),
-    destroy: deleteVet360Field('/profile/email_addresses', VET360_CONSTANTS.FIELD_NAMES.EMAIL, 'email')
-  },
-
-  [VET360_CONSTANTS.FIELD_NAMES.MAILING_ADDRESS]: {
-    update: updateVet360Field('/profile/addresses', VET360_CONSTANTS.FIELD_NAMES.MAILING_ADDRESS, 'address'),
-    destroy: deleteVet360Field('/profile/addresses', VET360_CONSTANTS.FIELD_NAMES.MAILING_ADDRESS, 'address')
-  },
-
-  [VET360_CONSTANTS.FIELD_NAMES.RESIDENTIAL_ADDRESS]: {
-    update: updateVet360Field('/profile/addresses', VET360_CONSTANTS.FIELD_NAMES.RESIDENTIAL_ADDRESS, 'address'),
-    destroy: deleteVet360Field('/profile/addresses', VET360_CONSTANTS.FIELD_NAMES.RESIDENTIAL_ADDRESS, 'address')
-  }
-};
```

Comment:
This was the bulk of the changes. This includes -

- The different formatters have been moved into the EmailField/AddressField/PhoneField components and used as a prop in `ProfileField`.
- The `update`/`destroy` functions for each property are derived in `ProfileField`.